### PR TITLE
Wikitext docs overhaul first step

### PR DESCRIPTION
**Pull request by pmario**
- The WikiText tiddler overview tiddler shows a logical sort order 
   - The list is ordered by complexity -- simple -> complex
   - The list is 4 columns and is responsive with 2 columns for small screens
- There is a button to open all rules in the StoryRiver
- "Formatting in WikiText" uses a table now and has more info
- ~~There is a new wikitext-template tiddler, that adds a footer to every tiddler tagged WikiText. The footer shows a tip about the tag-pill .. for new users, that don't know the tagging mechanism~~
- The PR moves the multi-colums CSS definition used in "Formatting List Results as Tables with CSS - Specified Columns Methods" to the $:/_tw5.com-styles tiddler, so it can be reused 
- MarkDown is a new definition tiddler. I don't want to have external links, when we explain our TW syntax.
- "Widgets in Wikitext" ... links to Widgets now, which shows the list of all widgets. There is no need to have that list while explaining the wikitext syntax.

There may be more steps, but this could be a start. This [discussion at Talk](https://talk.tiddlywiki.org/t/a-better-documentation-for-wikitext/4066) triggered the PR.

### CODE REVIEW

**Comment by pmario** [comment]
Json file to play wiht:  [wikitext-without-template.zip](https://github.com/Jermolene/TiddlyWiki5/files/9197541/wikitext-without-template.zip)

File: editions/tw5.com/tiddlers/concepts/WikiText.tid [hunk 1]
```diff
@@ -1,15 +1,20 @@
...
+modified: 20220727095329372
 tags: Concepts Reference
 title: WikiText
 type: text/vnd.tiddlywiki
 
-~WikiText is a concise, expressive way of typing a wide range of text formatting, hypertext and interactive features. It allows you to focus on writing without a complex user interface getting in the way. It is designed to be familiar for users of [[MarkDown|http://daringfireball.net/projects/markdown/]], but with more of a focus on linking and the interactive features.
+~WikiText is a concise, expressive way of typing a wide range of text formatting, hypertext and interactive features. It allows you to focus on writing without a complex user interface getting in the way. It is designed to be familiar for users of MarkDown but with more of a focus on linking and the interactive features.
```
Comment (Jermolene) [inline editions/tw5.com/tiddlers/concepts/WikiText.tid:8]: In fact the docs were wrong originally, Markdown does not have an uppercase "D"

File: editions/tw5.com/tiddlers/concepts/WikiText.tid [hunk 2]
```diff
@@ -1,15 +1,20 @@
...
 
-See [[Formatting text in TiddlyWiki]] for an introduction to WikiText.
+The following list of syntax rules ''start simple and get more complex'' at the end of the list.
 
-The following elements of WikiText syntax are built into the core:
+<$button actions=<<wikitext-open-custom-list>> >Open all tiddlers from the list below</$button>
 
-<<list-links "[tag[WikiText]]">>
+@@.multi-columns
+<<list-links-draggable tiddler:"WikiText" field:"custom-list">>
```
Comment (Jermolene) [inline editions/tw5.com/tiddlers/concepts/WikiText.tid:17]: Wouldn't it be more convenient to keep the list inline here? ie:

```
<<list-links-draggable "[[Headings in WikiText]] [[Paragraphs in WikiText]] [[Hard Linebreaks in WikiText]] [[Formatting in WikiText]] [[Dashes in WikiText]] [[Block Quotes in WikiText]] [[Lists in WikiText]] [[Definitions in WikiText]] [[Horizontal Rules in WikiText]] [[Tables in WikiText]] [[Code Blocks in WikiText]] [[Typed Blocks in WikiText]] [[Images in WikiText]] [[Linking in WikiText]] [[Variables in WikiText]] [[Macro Definitions in WikiText]] [[Macro Calls in WikiText]] [[Macros in WikiText]] [[Transclusion in WikiText]] [[Transclusion and Substitution]] [[Styles and Classes in WikiText]] [[HTML in WikiText]] [[Widgets in WikiText]] [[WikiText Parser Modes]]">>
```

File: editions/tw5.com/tiddlers/concepts/WikiText.tid [hunk 3]
```diff
@@ -1,15 +1,20 @@
...
 title: WikiText
 type: text/vnd.tiddlywiki
 
-~WikiText is a concise, expressive way of typing a wide range of text formatting, hypertext and interactive features. It allows you to focus on writing without a complex user interface getting in the way. It is designed to be familiar for users of [[MarkDown|http://daringfireball.net/projects/markdown/]], but with more of a focus on linking and the interactive features.
+~WikiText is a concise, expressive way of typing a wide range of text formatting, hypertext and interactive features. It allows you to focus on writing without a complex user interface getting in the way. It is designed to be familiar for users of MarkDown but with more of a focus on linking and the interactive features.
 
 ~WikiText can also be inserted to the text field using the [[Editor toolbar]].
 
-See [[Formatting text in TiddlyWiki]] for an introduction to WikiText.
+The following list of syntax rules ''start simple and get more complex'' at the end of the list.
```
Comment (Jermolene) [inline editions/tw5.com/tiddlers/concepts/WikiText.tid:12]: It is not necessary to use bold here. Please try to keep to plain, ordinary formatting and punctuation as much as possible.

In any case, better worded as "The following list of syntax rules starts with the simpler ones and moves on to the more complex ones."

File: editions/tw5.com/tiddlers/definitions/MarkDown.tid [hunk 4]
```diff
@@ -0,0 +1,8 @@
+created: 20220727090611178
+modified: 20220727095958361
+tags: Definitions
+title: MarkDown
```
Comment (Jermolene) [inline editions/tw5.com/tiddlers/definitions/MarkDown.tid:4]: Should be "Mardown" (see above)

File: editions/tw5.com/tiddlers/concepts/WikiText.tid [hunk 5]
```diff
@@ -1,15 +1,20 @@
...
 
-The following elements of WikiText syntax are built into the core:
+<$button actions=<<wikitext-open-custom-list>> >Open all tiddlers from the list below</$button>
```
Comment (Jermolene) [inline editions/tw5.com/tiddlers/concepts/WikiText.tid:14]: My point here was that we should not single out this tiddler for a special "open all children" button. If it is useful then we should add it consistently across all tiddlers, and not just for this one. In any case it should be a separate PR

**Review by Jermolene — COMMENTED** [review]
Thanks for your patience @pmario much appreciated

File: editions/tw5.com/tiddlers/concepts/WikiText.tid (hunk 2 above)
Comment (pmario) [reply editions/tw5.com/tiddlers/concepts/WikiText.tid:17]: Not really, since the "open all" button also needs it.

File: editions/tw5.com/tiddlers/concepts/WikiText.tid (hunk 1 above)
Comment (pmario) [reply editions/tw5.com/tiddlers/concepts/WikiText.tid:8]: will change that

File: editions/tw5.com/tiddlers/concepts/WikiText.tid (hunk 2 above)
Comment (Jermolene) [reply editions/tw5.com/tiddlers/concepts/WikiText.tid:17]: But I am suggesting that we remove the "open all" button

I reiterate that my goal here is to try to keep the docs as internally consistent as possible.

Comment (pmario) [reply editions/tw5.com/tiddlers/concepts/WikiText.tid:17]: [list-links-draggable](https://tiddlywiki.com/prerelease/#list-links-draggable%20Macro) needs a "place" to store the results. If new elements may be added in the future, we can simply "drag & drop" them to the list. .. Or the list can be re-sorted, if it turns out the order isn't 100% right yet.

File: editions/tw5.com/tiddlers/concepts/WikiText.tid (hunk 5 above)
Comment (pmario) [reply editions/tw5.com/tiddlers/concepts/WikiText.tid:14]: We constantly get feedback from new users, that strict alphabetical listing doesn't make sense for them and the "non linearity" is overwhelming at the beginning, because they don't know, where to start and where to end. 

That's why I did try to create a "User Story" by rearranging the list to have the "Basic topics" at the beginning and the "Advanced" at the second part of the list. 

image

If a new user wants to get an overview of all possible wikitext rules, they need the possibility to open the "User Story", in an order they can understand.

They can read the whole story from top to bottom and get info from simple to complex. There is no other way at the moment to accomplish this task in a convenient way. ... The "open all" button should be that way. ... May be it should be named "Open the WikiText user story"

>.. If it is useful then we should add it consistently across all tiddlers, and not just for this one. In any case it should be a separate PR

The `<<wikitext-open-custom-list>>` macro should be a consistent way to get "Story Lists". At the moment it uses the "custom-list" field to store that order. ... May be we should name it "user-story" 

This PR is part of the documentation branch. So it can be changed quickly. If it doesn't work out we can change it again. 

BUT if we don't change anything to get feedback from the community, we don't know, how we can improve the rest of the docs, so new users are comfortable with it. --- The only feedback we can get is: "It's way to complicated"

This PR should have been the "first step" to improve readability and approachability for new users. The [Overview](https://github.com/Jermolene/TiddlyWiki5/issues/6829) shows more topics, which I think could help us. It would add more interactivity, so users can try out the syntax while reading it, with 2 windows open side by side. 

If it is impossible to change the structure of the documentation. How should the community improve it? ... They could add one more thread at Talk, with a hypothetical discussion, how to improve the documentation.

May be we decide that Talk should be the medium to be used to document TiddlyWiki, using markdown instead of wikitext. .. But that's not what I want.

**Comment by pmario** [comment]
@Jermolene ... Sorry  ... I managed to completely mess up this PR by accidentally merging "master" to be able to pull the branch. I'll close it and will try one more. ... Hopefully "All good things go by three"
